### ISSUE TITLE
Add an `ignore` option to `no-unused-expressions`

### ISSUE THREAD

**Issue by eTorAken** [eslint/eslint#8075]
**What rule do you want to change?**
`no-unused-expressions`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option `ignore` which takes objects' names.

**Please provide some example code that this change will affect:**
```jsx
import { expect } from 'chai';
const output = 1;
expect(output).to.be.NaN;
```

**What does the rule currently do for this code?**
Throws a `no-unused-expressions` error on the last line.

**What will the rule do after it's changed?**
```jsx
/*eslint no-unused-expressions: ["error", { "ignore": ['expect'] }]*/
import { expect } from 'chai';
const output = 1;
expect(output).to.be.NaN;
// => no warning
```

**Comment by gyandeeps** [comment]
@eTorAken I think your are proposing an enhancement for the rule. Can you please answer all the questions form our [rule change request](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)

**Comment by eTorAken** [comment]
My bad, thanks for your answer @gyandeeps, I've edited my post.

**Comment by not-an-aardvark** [comment]
Note: This was previously suggested in https://github.com/eslint/eslint/issues/2102.

**Comment by zxqfox** [comment]
Don't forget about:
```
chai.should();

output.should.be.NaN;
```

It's also viable for https://github.com/shouldjs/should.js

**Comment by ihordiachenko** [comment]
@eTorAken I've made a small [plugin](https://github.com/ihordiachenko/eslint-plugin-chai-friendly) that overrides default `no-unused-expressions` rule and makes it friendly towards chai `expect` statements. I hope this will help you.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
